lib/pbench/common/results.py

```diff
@@ -0,0 +1,119 @@
+        if not self.tarball.exists():
+            raise FileNotFoundError(f"Tar ball '{self.tarball}' does not exist")
+        self.tarball_len = tarball_len
+        self.tarball_md5 = tarball_md5
+        self.config = config
+        server_rest_url = config.get("results", "server_rest_url")
+        tbname = urllib.parse.quote(self.tarball.name)
+        self.upload_url = f"{server_rest_url}/upload/{tbname}"
+        self.logger = logger
+
+    def copy_result_tb(self, token: str) -> None:
+        """Copies the tar ball from the agent to the configured server using upload
+        API.
+
+        Args
+            token -- a token which establishes that the caller is
+                authorized to make the PUT request on behalf of a
+                specific user.
+
+        Raises
+            RuntimeError     if a connection fails to be make to the server
+            FileUploadError  if the tar balls to upload properly
+
+        """
+        try:
+            debug_unittest = bool(self.config.get("pbench-server", "debug_unittest"))
+        except Exception:
+            debug_unittest = False
+
+        if debug_unittest:
+            self.mock_copy_result_tb(token)
+            return 0
+
+        headers = {
+            "Content-MD5": self.tarball_md5,
+            "Authorization": f"Bearer {token}",
+            "controller": self.controller,
+        }
+        with self.tarball.open("rb") as f:
+            try:
+                request = requests.Request(
+                    "PUT", self.upload_url, data=f, headers=headers
+                ).prepare()
+
+                # Per RFC 2616, a request must not contain both
+                # Content-Length and Transfer-Encoding headers; however,
+                # the server would like to receive the Content-Length
+                # header, but the requests package may opt to generate
+                # the Transfer-Encoding header instead...so, check that
+                # we got what we want before we send the request.  Also,
+                # confirm that the contents of the Content-Length header
+                # is what we expect.
+                assert (
+                    "Transfer-Encoding" not in request.headers
+                ), "Upload request unexpectedly contains a `Transfer-Encoding` header"
+                assert (
+                    "Content-Length" in request.headers
+                ), "Upload request unexpectedly missing a `Content-Length` header"
+                assert request.headers["Content-Length"] == str(self.tarball_len), (
+                    "Upload request `Content-Length` header contains {} -- "
+                    "expected {}".format(
+                        request.headers["Content-Length"], self.tarball_len
+                    )
+                )
+
+                response = requests.Session().send(request)
+                response.raise_for_status()
+                self.logger.info("File uploaded successfully")
```

Comment:
You just moved this, but I think this message is essentially meaningless as written; we should at least include the tarball name.